llvm/test/CodeGen/Hexagon/loop-idiom/lcssa.ll

```diff
@@ -1,4 +1,4 @@
-; RUN: opt -S -hexagon-loop-idiom < %s | opt -S -passes='loop(loop-deletion),gvn'
+; RUN: opt -mtriple hexagon-- -S -passes='loop(hexagon-loop-idiom,loop-deletion),gvn'
 ; REQUIRES: asserts
 
 ; This tests that the HexagonLoopIdiom pass does not mark LCSSA information
```

Comment:
this is an issue. the new PM loop infra assumes that loops are always in LCSSA form. either the pass needs to ensure the loops are in LCSSA form, or perhaps it can just call a helper method

Reply:
I'll push a patch to make sure LCSSA form is preserved.